### PR TITLE
runtime: write oom file to notify CRI-O OOM occurred

### PR DESCRIPTION
**Pull request by liubin**
Fixes: #2955

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

File: containerd-shim-v2/wait.go [hunk 1]
```diff
@@ -151,6 +153,26 @@ func watchOOMEvents(ctx context.Context, s *service) {
...
+					"sandbox":   s.sandbox.ID(),
+					"container": containerID,
+				}).Infof("write oom file to notify CRI-O: %s", oomPath)
+
+				f, err := os.OpenFile(oomPath, os.O_CREATE, 0666)
+				if err != nil {
+					logrus.WithFields(logrus.Fields{
+						"sandbox":   s.sandbox.ID(),
+						"container": containerID,
+					}).Warnf("failed to write oom file %s", oomPath)
```
Comment (jodh-intel) [inline containerd-shim-v2/wait.go:169]: Nit: You could put `oomPath` in a field too.

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

**Review by fidencio — APPROVED** [review]
lgtm

**Comment by jodh-intel** [comment]
Hi @liubin - does this need porting to 2.0?

File: containerd-shim-v2/wait.go [hunk 2]
```diff
@@ -151,6 +153,26 @@ func watchOOMEvents(ctx context.Context, s *service) {
 				continue
 			}
 
+			// write oom file for CRI-O
+			if c, ok := s.containers[containerID]; ok && oci.IsCRIOContainerManager(c.spec) {
```
Comment (amshinde) [inline containerd-shim-v2/wait.go:157]: @liubin I am a bit skeptical about adding code checking for specific container manager. Can we simply create this file irrespective of what the upper stack looks like?
cc @bergwolf

**Pull request merged**